dual_net.py

```diff
@@ -307,22 +367,44 @@ def export_model(working_dir, model_path):
         tf.gfile.Copy(filename, destination_path)
 
 
-def train(working_dir, tf_records, steps=None):
-    estimator = get_estimator(working_dir)
-
-    def input_fn():
-        return preprocessing.get_input_tensors(
-            FLAGS.train_batch_size, tf_records, filter_amount=1.0,
-            shuffle_buffer_size=FLAGS.shuffle_buffer_size)
-
-    update_ratio_hook = UpdateRatioSessionHook(working_dir)
-    step_counter_hook = EchoStepCounterHook(output_dir=working_dir)
+def train(working_dir, *tf_records, steps=None):
+    if FLAGS.use_tpu:
+        tpu_cluster_resolver = tf.contrib.cluster_resolver.TPUClusterResolver(
+            FLAGS.tpu_name, zone=None, project=None)
+
+        config = tpu_config.RunConfig(
+            cluster=tpu_cluster_resolver,
+            model_dir=working_dir,
+            save_checkpoints_steps=max(600, FLAGS.iterations_per_loop),
+            tpu_config=tpu_config.TPUConfig(
+                iterations_per_loop=FLAGS.iterations_per_loop,
+                num_shards=FLAGS.num_cores,
+                per_host_input_for_training=tpu_config.InputPipelineConfig.PER_HOST_V2))  # pylint: disable=line-too-long
+
+        estimator = tpu_estimator.TPUEstimator(
+            use_tpu=FLAGS.use_tpu,
+            model_fn=model_fn,
+            config=config,
+            train_batch_size=FLAGS.train_batch_size,
+            eval_batch_size=FLAGS.train_batch_size)
+        def input_fn(params):
+            return preprocessing.get_tpu_input_tensors(params['batch_size'], tf_records)
+        hooks = []
```

Comment:
no step counter output?

Reply:
Yeah, it broke tpu stuff, will leave a todo to reintegrate.